### PR TITLE
Time Consistency

### PR DESCRIPTION
**Pull request by vEpiphyte**
Time is a flat circle. Everything we have done or will do we will do over and over and over again- forever.

### CODE REVIEW

File: synapse/lib/types.py [hunk 1]
```diff
@@ -814,41 +833,31 @@ def _normPyStr(self, valu):
         if valu == '?':
             raise s_exc.BadTypeValu(name=self.name, valu=valu, mesg='interval requires begin time')
 
-        norm, _ = self.timetype.norm(valu)
-
-        return (norm, norm + 1), {}
+        minv, _ = self.timetype.norm(valu)
+        # Norm is gauranteed to be a valid time value, but norm +1 may not be
```
Comment (jnwatson) [inline synapse/lib/types.py]: speeling: guaranteed

Comment (invisig0th) [inline synapse/lib/types.py]: speeeling :D

File: synapse/tests/test_lib_base.py [hunk 2]
```diff
@@ -331,3 +331,28 @@ def test_base_main_sigint(self):
 
         proc.join(timeout=10)
         self.eq(proc.exitcode, 137)
+
+    async def test_onwith(self):
```
Comment (invisig0th) [inline synapse/tests/test_lib_base.py]: haha, just couldn't resist it :D  I was wondering if i'd see this in it's own PR or slipped into another ;)

File: synapse/tests/test_lib_types.py [hunk 3]
```diff
@@ -818,34 +842,46 @@ def test_syntag(self):
 
             async with await core.snap() as snap:
                 node = await snap.addNode('test:str', 'a', {'tick': '2014'})
+                created = node.get('.created')
+                await asyncio.sleep(0.01)
```
Comment (invisig0th) [inline synapse/tests/test_lib_types.py]: what's this about?

**Pull request merged**